smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/HttpQueryTraitValidator.java

```diff
@@ -77,8 +85,39 @@ private List<ValidationEvent> validateBindings(Map<StructureShape, Map<String, S
                             paramsToMembers.getKey(), ValidationUtils.tickedList(paramsToMembers.getValue()))));
                 }
             }
+
+            List<OperationShape> operations = structureToOperations.getOrDefault(entry.getKey(),
+                                                                                 Collections.emptyList());
+            for (OperationShape operation : operations) {
+                UriPattern pattern = operation.expectTrait(HttpTrait.class).getUri();
+                for (Map.Entry<String, String> literalEntry : pattern.getQueryLiterals().entrySet()) {
+                    String literalKey = literalEntry.getKey();
+                    if (entry.getValue().containsKey(literalKey)) {
+                        events.add(error(entry.getKey(), String.format(
+                            "`httpQuery` parameter name binding conflicts found for the `%s` parameter, the "
+                            + "http trait for the `%s` operation defines a query literal with the same name",
+                            literalKey, operation.getId())));
+                    }
+                }
+            }
         }
 
         return events;
     }
+
+    private Map<StructureShape, List<OperationShape>> getStructureToOperations(Model model) {
+        Map<StructureShape, List<OperationShape>> structureToOperations = new HashMap<>();
+        for (ServiceShape service : model.getServiceShapes()) {
+            for (OperationShape operation : TopDownIndex.of(model).getContainedOperations(service)) {
```

Comment:
My understanding is that _any_ operation in the model with the http trait could have this issue, so why do we need to traverse the model on a service-by-service basis? 
Then we could get rid of the outer loop and just do 
```java
for (OperationShape operation : model.getOperationShapesWithTrait(HttpTrait.class)) {
   ...
}
```